.evergreen/config.yml

```diff
@@ -190,6 +190,10 @@ functions:
           if [ "${is_patch}" = "true" ]; then
             echo "Patch build detected, skipping"
           fi
+          if [ "${branch}" != "master" ]; then
+            echo "publish snapshot is only run on master branch."
+            echo "Detected branch '${branch}', skipping."
+          fi
```

Comment:
You will also need to add a check in the next section:

```
           if [ "${is_patch}" = "true" ]; then
             exit 0
           fi
+          if [ "${branch}" != "master" ]; then
+            exit 0
+          fi
```

That said, it's not obvious to me why there is a check with a message in one `command` section and then the same check with the `exit` in the next `command` section.  These two sections could probably be merged, but that's up to you.

Reply:
Good catch. Updated. The sections cannot be merged since the second `shell.exec` has `silent: true`. I assume the reason is to prevent logging credentials.

Reply:
Ah, that makes sense why it is in two separate sections.